lib/pure/parseopt.nim

```diff
@@ -192,17 +192,15 @@ proc parseWord(s: string, i: int, w: var string,
       add(w, s[result])
       inc(result)
 
-when declared(os.paramCount):
-  # we cannot provide this for NimRtl creation on Posix, because we can't
-  # access the command line arguments then!
-
+when declared(os.paramCount) or (NimMajor, NimMinor, NimPatch) >= (1, 5, 1):
   proc initOptParser*(cmdline = "", shortNoVal: set[char] = {},
                       longNoVal: seq[string] = @[];
                       allowWhitespaceAfterColon = true): OptParser =
     ## Initializes the command line parser.
     ##
     ## If ``cmdline == ""``, the real command line as provided by the
```

Comment:
pre-existing but this creates a weird edge case

Reply:
see https://github.com/nim-lang/Nim/pull/17009#discussion_r574942695 for details. It's buggy.